source/extensions/filters/network/dubbo_proxy/hessian_utils.cc

```diff
@@ -0,0 +1,530 @@
+#include "extensions/filters/network/dubbo_proxy/hessian_utils.h"
+
+#include <type_traits>
+
+#include "common/common/assert.h"
+#include "common/common/fmt.h"
+
+#include "extensions/filters/network/dubbo_proxy/buffer_helper.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace DubboProxy {
+
+namespace {
+
+template <typename T>
+typename std::enable_if<std::is_signed<T>::value, T>::type leftShift(T left, uint16_t bit_number) {
+  if (left < 0) {
+    left = left * -1;
+    return -1 * (left << bit_number);
+  }
+
+  return left << bit_number;
+}
+
+} // namespace
+
+char* writeIntoString(std::string* str, size_t length) {
+  str->reserve(length);
```

Comment:
no need this, also per @jmarantz comment on previous PR this function could have better name.

Reply:
I think it's necessary to reserve, is to avoid allocating memory for many times, if use the resize, can actually lead to allocate memory for many times. because the resize is append directly, using the default memory allocation strategy   

[resize](https://github.com/gcc-mirror/gcc/blob/master/libstdc%2B%2B-v3/include/bits/basic_string.tcc#L352)

Reply:
If I read that source correctly, `resize` call `append` once and it calls `reserve` at most once.

https://github.com/gcc-mirror/gcc/blob/master/libstdc%2B%2B-v3/include/bits/basic_string.tcc#L758

Reply:
Your understanding is right, but the reserve here still can't get rid of, my code is a little mistake, here I revised once, this is the reference of chromium WriteInfo

https://cs.chromium.org/chromium/src/base/strings/string_util.h?q=WriteInto&sq=package:chromium&dr=CSs&l=426

https://cs.chromium.org/chromium/src/base/strings/string_util.cc?sq=package:chromium&dr=CSs&g=0&l=908

Reply:
I think you broke the test (esp. ASAN), the length you passed below doesn't contain the null end.